infra/base-images/base-builder/jcc/jcc.go

```diff
@@ -330,61 +73,18 @@ func main() {
 	args := os.Args[1:]
 	basename := filepath.Base(os.Args[0])
 	isCPP := basename == "clang++-jcc"
-	newArgs := append(args, "-w")
+	newArgs := args
 
 	var bin string
 	if isCPP {
 		bin = "clang++"
-		newArgs = append(args, "-stdlib=libc++")
 	} else {
 		bin = "clang"
 	}
- 	fullCmdArgs := append([]string{bin}, newArgs...)
- 	retcode, out, errstr := Compile(bin, newArgs)
- 	if retcode == 0 {
- 	 	WriteStdErrOut(fullCmdArgs, out, errstr)
- 	 	os.Exit(0)
- 	}
-
- 	// Some projects convert warnings to errors, undo this and try again.
- 	newArgs = RemoveFailureCausingFlags(newArgs)
- 	retcode, out, errstr = Compile(bin, newArgs)
- 	fullCmdArgs = append([]string{bin}, newArgs...)
- 	if retcode == 0 {
- 	 	WriteStdErrOut(fullCmdArgs, out, errstr)
- 	 	os.Exit(0)
- 	}
-
-	// Note that on failures or when we succeed on the first try, we should
-	// try to write the first out/err to stdout/stderr.
-	// When we fail we should try to write the original out/err and one from
-	// the corrected.
-
-	headersFixArgs, headersFixed, _ := CorrectMissingHeaders(bin, newArgs)
-	if headersFixed {
-		// We succeeded here but it's kind of complicated to get out and
-		// err from TryCompileAndFixHeadersOnce. The output and err is
-		// not so important on success so just be silent.
-		WriteStdErrOut(append([]string{bin}, headersFixArgs...), "", "")
-		os.Exit(0)
-	}
-
-	if isCPP {
-		// Nothing else we can do. Just write the error and exit.
-		// Just print the original error for debugging purposes and
-		//  to make build systems happy.
-		WriteStdErrOut(fullCmdArgs, out, errstr)
-		os.Exit(retcode)
-	}
-	fixargs, fixret, fixout, fixerr := TryFixCCompilation(newArgs)
-	if fixret != 0 {
-		// We failed, write stdout and stderr from the first failure and
-		// from fix failures so we can know what the code did wrong and
-		// how to improve jcc to fix more issues.
+	fullCmdArgs := append([]string{bin}, newArgs...)
+	retcode, out, errstr := Compile(bin, newArgs)
+	if retcode == 0 {
 		WriteStdErrOut(fullCmdArgs, out, errstr)
-		fmt.Println("\nFix failure")
-		os.Exit(retcode)
+		os.Exit(0)
```

Comment:
rather than only calling WriteStdErrOut and os.Exit() when retcode == 0, I think we need to do this in all cases?

Reply:
Good point, fixed.